File: app/src/main/java/kztproject/jp/splacounter/api/MiniatureGardenClient.java
package kztproject.jp.splacounter.api;

import javax.inject.Inject;

import io.reactivex.Single;
import kztproject.jp.splacounter.model.Counter;
import kztproject.jp.splacounter.util.GameCountUtils;

public class MiniatureGardenClient {

    private final MiniatureGardenService service;

    @Inject
    public MiniatureGardenClient(MiniatureGardenService service) {
        this.service = service;
    }

    public Single<Counter> getCounter(int userId) {
        return service.getCounter(userId);
    }

    public Single<Counter> consumeCounter(int userId) {
        return service.cosumeCounter(userId, GameCountUtils.GAME_UNIT);
    }

    public Single<Counter> consumeCounter(int userId, int point) {
        return service.cosumeCounter(userId, GameCountUtils.GAME_UNIT * point);
    }
}

File: app/src/main/java/kztproject/jp/splacounter/api/MiniatureGardenClient.kt
package kztproject.jp.splacounter.api

import io.reactivex.Single
import kztproject.jp.splacounter.model.Counter
import kztproject.jp.splacounter.util.GameCountUtils
import javax.inject.Inject

class MiniatureGardenClient @Inject
constructor(private val service: MiniatureGardenService) {

    fun getCounter(userId: Long): Single<Counter> {
        return service.getCounter(userId.toInt())
    }

    fun consumeCounter(userId: Long, point: Int): Single<Counter> {
        return service.consumeCounter(userId.toInt(), GameCountUtils.GAME_UNIT * point)
    }
}
